element-connector/src/main/java/org/eclipse/californium/elements/RawData.java

```diff
@@ -225,13 +227,15 @@ public static RawData inbound(byte[] data, InetSocketAddress address, Principal
 	 * 
 	 * @param data the data to send.
 	 * @param address the IP address and port the data is to be sent to.
+	 * @param correlationContext correlation context for sending data (may be <code>null</code>).
 	 * @param callback the handler to call when this message has been sent (may be <code>null</code>).
 	 * @param useMulticast indicates whether the data should be sent using a multicast message.
 	 * @return the raw data object containing the outbound message.
 	 * @throws NullPointerException if data or address is <code>null</code>.
 	 */
-	public static RawData outbound(byte[] data, InetSocketAddress address, MessageCallback callback, boolean useMulticast) {
+	public static RawData outbound(byte[] data, InetSocketAddress address, CorrelationContext correlationContext, MessageCallback callback, boolean useMulticast) {
```

Comment:
The `callback` param is supposed to be used by the connector to set the `CorrelationContext` on an outbound request. How does that fit with providing the `CorrelationContext` in the `correlationContext` param?
Can you provide values for both params? If not, we should introduce another factory method accepting an existing correlation context instead of a callback, shouldn't we?